src/lib/resourceFilters.js

```diff
@@ -0,0 +1,362 @@
+function getTumorSize(bundle) {
+  const metaProfiledResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.where(meta.profile = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-tumor-size')",
+  );
+  const constrainedResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.ofType(Observation).where(code.coding.code = '21889-1' and code.coding.system = 'http://loinc.org')",
+  );
+  return mergeWithoutDuplicates(metaProfiledResources, constrainedResources);
+}
+// Specimens with Specimen.type of `TUMOR` from system http://terminology.hl7.org/CodeSystem/v2-0487 must be TumorSpecimens: https://hl7.org/fhir/us/mcode/StructureDefinition-mcode-tumor-specimen.html#conformance
+function getTumorSpecimen(bundle) {
+  const metaProfiledResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.where(meta.profile = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-tumor-specimen')",
+  );
+  const constrainedResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.ofType(Specimen).where(type.coding.code = 'TUMOR' and type.coding.system = 'http://terminology.hl7.org/CodeSystem/v2-0487')",
+  );
+  return mergeWithoutDuplicates(metaProfiledResources, constrainedResources);
+}
+
+// Disease
+// Observations with a code from an extensible valueset, TumorMarkerTestVS, must be Tumor Marker Tests: https://hl7.org/fhir/us/mcode/StructureDefinition-mcode-tumor-marker-test.html#conformance
+function getTumorMarkerTest(bundle) {
+  const metaProfiledResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.where(meta.profile = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-tumor-marker-test')",
+  );
+  const constrainedResources = fhirpath
+    .evaluate(bundle, 'Bundle.entry.resource.ofType(Observation)')
+    .filter(
+      (observation) =>
+        observation.code?.coding?.some((coding) =>
+          vsChecker.checkCodeInVs(coding.code, coding.system, 'TumorMarkerTestVS'),
+        ) ?? false,
+    );
+  return mergeWithoutDuplicates(metaProfiledResources, constrainedResources);
+}
+// Conditions with a code from an the extensible valueset, SecondaryCancerDisorderVS, must be Secondary Cancer Conditions: https://hl7.org/fhir/us/mcode/StructureDefinition-mcode-secondary-cancer-condition.html#conformance
+function getSecondaryCancerCondition(bundle) {
+  const metaProfiledResources = fhirpath.evaluate(
+    bundle,
+    "Bundle.entry.resource.where(meta.profile = 'http://hl7.org/fhir/us/mcode/StructureDefinition/mcode-secondary-cancer-condition')",
+  );
+  const constrainedResources = fhirpath
+    .evaluate(bundle, 'Bundle.entry.resource.ofType(Condition)')
+    .filter(
+      (condition) =>
+        condition.code?.coding?.some((coding) =>
+          vsChecker.checkCodeInVs(coding.code, coding.system, 'SecondaryCancerDisorderVS'),
+        ) ?? false,
+    );
+  return metaProfiledResources.length > 0 ? metaProfiledResources : constrainedResources;
```

Comment:
Missed one?

Reply:
Yep 😅. Gonna fix this in the next commit. Also gonna update our unit tests to test everything a bit more throughly.